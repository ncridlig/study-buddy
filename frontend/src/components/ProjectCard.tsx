'use client';

// ✅ Import CardActions for better button layout
import { Card, CardContent, Typography, Button, CardActions } from '@mui/material';
import Link from 'next/link';

interface Project {
  id: string;
  title: string;
  description: string;
}

// ✅ Updated props to include the onDelete function
interface ProjectCardProps {
  project: Project;
  onDelete: (id: string) => void;
}

// ✅ This component now accepts the onDelete prop
export default function ProjectCard({ project, onDelete }: ProjectCardProps) {
  return (
    // Added flex styles to ensure cards have equal height in a row
    <Card sx={{ display: 'flex', flexDirection: 'column', height: '100%' }}>
      <CardContent sx={{ flexGrow: 1 }}>
        <Typography variant="h6">{project.title}</Typography>
        <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
          {project.description}
        </Typography>
      </CardContent>

      {/* ✅ Added CardActions to neatly group the buttons */}
      <CardActions sx={{ justifyContent: 'space-between', p: 2 }}>
        <Button
          href={`/dashboard/projects/${project.id}`}
          component={Link}
          size="small"
        >
          Open Project
        </Button>
        {/* ✅ NEW: The Delete Button */}
        <Button
          size="small"
          color="error" // 'error' color makes it red, indicating a destructive action
          onClick={() => onDelete(project.id)}
        >
          Delete
        </Button>
      </CardActions>
    </Card>
  );
}