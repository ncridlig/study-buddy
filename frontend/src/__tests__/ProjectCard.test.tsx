// __tests__/ProjectCard.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import ProjectCard from '@/components/ProjectCard';
import '@testing-library/jest-dom';

describe('ProjectCard', () => {
  const mockProject = {
    id: 'abc123',
    title: 'AI Research',
    description: 'Exploring GPT models and LLMs',
  };

  it('renders project title and description', () => {
    render(<ProjectCard project={mockProject} />);
    
    expect(screen.getByText('AI Research')).toBeInTheDocument();
    expect(screen.getByText('Exploring GPT models and LLMs')).toBeInTheDocument();
  });

  it('renders a button linking to the project detail page', () => {
    render(<ProjectCard project={mockProject} />);
    
    const link = screen.getByRole('link', { name: /open project/i });
    expect(link).toHaveAttribute('href', '/dashboard/projects/abc123');
  });

  it('matches snapshot', () => {
    const { container } = render(<ProjectCard project={mockProject} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
