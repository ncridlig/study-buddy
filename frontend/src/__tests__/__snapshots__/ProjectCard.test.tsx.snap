// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProjectCard matches snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
  style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
    >
      AI Research
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body2 css-tmy7vb-MuiTypography-root"
    >
      Exploring GPT models and LLMs
    </p>
    <a
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-q7p70v-MuiButtonBase-root-MuiButton-root"
      href="/dashboard/projects/abc123"
      tabindex="0"
    >
      Open Project
    </a>
  </div>
</div>
`;
